Apps/WebClient/src/Server/Listeners/BannerListener.cs

```diff
@@ -74,40 +75,52 @@ public class BannerListener : BackgroundService
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "Abstract class property")]
         protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
-            this.logger.LogDebug($"Registering Channel Notification on channel {Channel}");
+            this.logger.LogInformation("DBChangeListener is starting");
             stoppingToken.Register(() =>
-                this.logger.LogInformation($"DBChangeListener exiting..."));
-            try
+                this.logger.LogInformation($"DBChangeListener Shutdown as cancellation requested    "));
+            int attempts = 0;
+            while (!stoppingToken.IsCancellationRequested)
```

Comment:
I have a feeling that this needs some improvements but not sure how. Should we invest a bit further to see if there is another way to do this without 2 whiles?

Reply:
Yeah when I first wrote it I thought it strange but the two loops are handling different things and just ensuring that we exit properly if a shutdown request comes in.  The outer loop is ensuring that we rebuild our listener and the inner loop is waiting on events from the DB.